### DISCUSSION TITLE
Is it possible to intercept the connection and stop it occuring?

### DISCUSSION THREAD

**Discussion by javaspeak** [MetaMask/snaps#1805] [Q&A]
I want to do the following:

(i) Intercept the connection to the wallet
(ii) Make a call to an external API to get feedback on the wallet address
(iii) If the feedback is negative block the connection to the wallet with a suitable warning message.

Is that possible in a snap?

**Comment by Montoya** [comment]
This is not possible as of yet

**Reply by databu** [reply]
Any plans to make it possible? Or this this something that would not be wanted?

**Reply by Montoya** [reply]
It's not like we have ruled it out, it could be possible at some point. One way to start the process of making the feature a reality is to submit a Snaps Improvement Proposal: https://github.com/MetaMask/SIPs#readme

**Reply by databu** [reply]
Will do, thanks

**Reply by javaspeak** [reply]
@Montoya I raised an issue to create a SIP:

https://github.com/MetaMask/SIPs/issues/108

Can someone add me to the repo so I can: create a branch for the SIP; set it to Draft and then create a merge request?

**Reply by Montoya** [reply]
Hi, please fork the repo and create your branch + PR from the fork
